Example/src/App.js

```diff
@@ -127,34 +117,46 @@ export function MainScreenItem({ item, onPressItem, screens }) {
   );
 }
 
-function LaunchReanimated1({ navigation }) {
+function LaunchReanimated1({ setUseRea2 }) {
   return (
     <>
       <ItemSeparator />
-      <RectButton
-        style={styles.button}
-        onPress={() => navigation.navigate('Reanimated1')}>
+      <RectButton style={styles.button} onPress={() => setUseRea2?.(false)}>
         <Text style={styles.buttonText}>👵 Reanimated 1.x Examples</Text>
       </RectButton>
     </>
   );
 }
 
-const Reanimated2App = createStackNavigator(
-  {
-    Main: { screen: MainScreen },
-    ...SCREENS,
-  },
-  {
-    initialRouteName: 'Main',
-    headerMode: 'screen',
-  }
+const Stack = createStackNavigator();
+
+const Reanimated2 = (setUseRea2) => (
+  <Stack.Navigator>
+    <Stack.Screen
+      name="Home"
+      options={{ title: '🎬 Reanimated 2.x Examples' }}
+      children={(props) => <MainScreen {...props} setUseRea2={setUseRea2} />}
+    />
+    {Object.keys(SCREENS).map((name) => (
+      <Stack.Screen
+        key={name}
+        name={name}
+        getComponent={() => SCREENS[name].screen}
+        options={{ title: SCREENS[name].title || name }}
+      />
+    ))}
+  </Stack.Navigator>
 );
 
-const ExampleApp = createSwitchNavigator({
-  Reanimated2App,
-  Reanimated1,
-});
+export default function App() {
+  const [useRea2, setUseRea2] = React.useState(true);
+
+  return (
+    <NavigationContainer>
+      {useRea2 ? Reanimated2(setUseRea2) : Reanimated1(setUseRea2)}
```

Comment:
Why do we need to specify if we're using Rea v1 or v2?

Reply:
It is a state used to switch between Reanimated2 and Reanimated1 examples, something like in auth flow: https://reactnavigation.org/docs/auth-flow.